ci/check-protobuf-stability.sh

```diff
@@ -3,6 +3,17 @@
 # SPDX-License-Identifier: Apache-2.0
 set -euo pipefail
 
+# The `SYSTEM_PULLREQUEST_TARGETBRANCH` environment variable is defined by
+# Azure Pipelines; in order to run this script locally, define it beforehand
+# as the branch being targeted. For example:
+#
+#   SYSTEM_PULLREQUEST_TARGETBRANCH=main bash -x ci/check-protobuf-stability.sh
+#
+TARGET="${SYSTEM_PULLREQUEST_TARGETBRANCH:-main}"
+echo "The target branch is '${TARGET}'."
+
+BUF_TAG_TO_CHECK=""
```

Comment:
I don't think this line does anything: the value is overridden through all code paths anyway, unless I'm missing something.

Reply:
Yes, it is overridden but it's declared here just for visibility. Do you prefer we don't declare the vars beforehand?

Reply:
I usually don't; the whole point of `set -u` is to detect if we forget to set a var before using it, and this seems like it foils that: we'll happily run with an empty string instead of properly failing.

Reply:
This seems not having been addressed yet, @nicu-da do you agree with @garyverhaegen-da (I do)?

Reply:
Done